frontend/src/scenes/data-warehouse/editor/multitabEditorLogic.tsx

```diff
@@ -86,25 +86,37 @@ const getNextUntitledNumber = (tabs: QueryTab[]): number => {
 }
 
 const getStorageItem = async (key: string): Promise<string | null> => {
-    const dbValue = await get(key)
-
-    if (dbValue) {
-        return dbValue
-    }
-
     const lsValue = localStorage.getItem(key)
-
     if (lsValue) {
-        await set(key, lsValue)
-        localStorage.removeItem(key)
+        try {
+            await set(key, lsValue)
+            localStorage.removeItem(key)
+        } catch (error) {
+            // this can happen if the storage quota is exceeded, or if the user is in private mode when trying to migrate
+            posthog.capture('sql-editor-local-storage-migration-failure', { error })
+        }
         return lsValue
     }
-
+    try {
+        const dbValue = await get(key)
+        if (dbValue) {
+            return dbValue
+        }
+    } catch (error) {
+        // this can happen if the storage quota is exceeded, or if the user is in private mode when trying to read from IndexedDB
+        posthog.capture('sql-editor-indexeddb-read-failure', { error })
+    }
     return null
 }
 
 const setStorageItem = async (key: string, value: string): Promise<void> => {
-    await set(key, value)
+    try {
+        await set(key, value)
+    } catch (error) {
+        // this would trigger if theres a problem writing to IndexedDB, and therefore we fall back to localStorage
+        posthog.capture('sql-editor-indexeddb-write-failure', { error })
```

Comment:
We should be using `posthog.captureException(..)` here instead so that these go into error tracking